src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1379,6 +1384,20 @@ static int16_t ConvertLockType(int16_t managedLockType)
     }
 }
 
+int64_t SystemNative_GetFileSystemType(intptr_t fd)
+{
+#if HAVE_STATFS_VFS || HAVE_STATFS_MOUNT
+    int statfsRes;
+    struct statfs statfsArgs;
+    // for our needs (get file system type) statfs is always enough and there is no need to use statfs64
+    // which got deprecated in macOS 10.6, in favor of statfs
+    while ((statfsRes = fstatfs(ToFileDescriptor(fd), &statfsArgs)) == -1 && errno == EINTR) ;
```

Comment:
I see this in the man page:
```
LSB has deprecated the library calls statfs() and fstatfs() and tells us to use statvfs(2) and fstatvfs(2) instead.
```

Reply:
From https://stackoverflow.com/a/14611791:

> If you just want file system capacity and usage information, the other answers are correct: prefer statvfs because it is standard POSIX and handles large file sizes better. statfs is BSD- and Linux-specific, with different structures on each. (Linux 2.6 added new statfs64 and fstatfs64 system calls that use an expanded structure to handle larger sizes.) However, statfs is still useful on Linux for determining the file system type (assuming you're okay with writing Linux-specific code).

> though I don't think fstatvfs provides file system type information.

yes, that is why I believe it's OK to use it here. If we ever want capacity information, we should use `statvfs `